src/BitcoinHelpers.js

```diff
@@ -698,39 +700,82 @@ const BitcoinHelpers = {
       expectedValue,
       outpoint
     ) {
-      const unspentTransactions = await electrumClient.getUnspentToScript(
+      const transactionsWithScript = await electrumClient.getTransactionsForScript(
         receiverScript
       )
 
-      for (const tx of unspentTransactions.reverse()) {
-        if (tx.value !== expectedValue) {
-          // Skip if the value doesn't match.
-          continue
+      /** @type TransactionInBlock[] */
+      const transactions = []
+
+      transactionsWithScript.forEach(tx => {
```

Comment:
This is all mutatey and stuff. I'd rather write it in terms of `map` and a series of transformations fwiw, though it's not worth going through the trouble right this moment. It would look something like:

```
transactionsWithScript
    .map(tx => Object.assign({}, tx, { confirmations: tx.confirmations || 0, value: tx.value * ... })
    .sort((txA, txB) => txB.confirmations - txA.confirmations || txB.value - txA.value)
    .filter(({ vin, vout }) => {
      const foundOutpoint = vout.find({ scriptPubKey, value }) =>
        scriptPubKey.hex === receiverScript && value >= expectedValue
      )
      return foundOutpoint && vin.find(({ txid, vout }) => BitcoinHelpers.compareOutpoint(foundOutpoint, BitcoinHelpers.outpointFrom(txid, vout)))
    }
    .map(...) // convert to the final format
```

Leaving this for probably myself haha.